scripts/prepare_redpajama.py

```diff
@@ -0,0 +1,183 @@
+import json
+import glob
+from multiprocessing import Pool
+import os
+from pathlib import Path
+import sys
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+import numpy as np
+from tqdm import tqdm
+
+from lit_llama import Tokenizer
+import lit_llama.indexed_dataset as indexed_dataset
+
+
+filenames_sample = [
+    "arxiv_sample.jsonl",
+    "book_sample.jsonl",
+    "c4_sample.jsonl",
+    "cc_2019-30_sample.jsonl",
+    "cc_2020-05_sample.jsonl",
+    "cc_2021-04_sample.jsonl",
+    "cc_2022-05_sample.jsonl",
+    "cc_2023-06_sample.jsonl",
+    "github_sample.jsonl",
+    "stackexchange_sample.jsonl",
+    "wikipedia_sample.jsonl",
+]
+
+filename_sets = {
+    "arxiv": "arxiv/arxiv*",
+    "book": "book/book*",
+    "c4": "c4/c4-train*",
+    "common_crawl": "common_crawl/*",
+    "github": "github/filtered*",
+    "stackexchange": "stackexchange/stackexchange*",
+    "wikipedia": "wikipedia/wiki*",
+}
+
+
+def prepare_sample(
+    source_path: Path,
+    tokenizer_path: Path,
+    destination_path: Path,
+    match = ""
+) -> None:
+    """Prepare the "Red Pajama" dataset. We assume tokenizer has been trained (i.e. we reuse LLaMA's tokenizer model)."""
+    destination_path.mkdir(parents=True, exist_ok=True)
+
+    tokenizer = Tokenizer(tokenizer_path)
+
+    for name in filenames_sample:
+        if match and match not in name:
+            continue
+
+        filepath = source_path / name
+
+        if not filepath.is_file():
+            raise RuntimeError(
+                f"Input file not found at {filepath}. \n"
+                "Make sure you download the data, e.g. wget -i https://data.together.xyz/redpajama-data-1T/v1.0.0/urls.txt or through \n"
+                "https://huggingface.co/datasets/togethercomputer/RedPajama-Data-1T \n"
+                "https://huggingface.co/datasets/togethercomputer/RedPajama-Data-1T-Sample \n"
+            )
```

Comment:
For me, this error didn't trigger. It just generated empty files in the output folder data/red_pajama_sample/.

Reply:
If glob didn't return files, this for loop will never run and then the `if not filepath.is_file():` check will never trigger

Reply:
Updated